src/main/java/parameter_service_demo/service/ParameterService.java

```diff
@@ -31,17 +36,45 @@ public ParameterDto updateById(Long id, NewParameterDto newParameterDto) {
         if (!parameterRepository.existsById(id)) {
             throw new EntityNotFoundException("Parameter", id);
         }
+        validateActive(id);
 
         var parameterEntity = newParameterDtoToEntity(newParameterDto);
         parameterEntity.setId(id);
         return parameterEntityToDto(parameterRepository.save(parameterEntity));
     }
 
+    public List<ParameterDto> activateByName(String name, LocalDate activationDate) {
+        return activate(name, activationDate).stream().map(this::parameterEntityToDto).toList();
+    }
+
     @Transactional
     public void deleteById(Long id) {
+        validateActive(id);
         parameterRepository.deleteById(id);
     }
 
+    @Transactional
+    protected List<ParameterEntity> activate(String name, LocalDate activationDate) {
+        var parameterEntities = parameterRepository.findByName(name);
+
+        for (var parameterEntity : parameterEntities) {
+            if (parameterEntity.isActive()) {
+                throw new ImmutableParameterChangeException(parameterEntity.getId());
+            }
+            parameterEntity.setActiveFrom(activationDate);
```

Comment:
If we get active parameter here, other parameters with the same name won't activate